functions/processQueueItem-test.js

```diff
@@ -72,14 +68,42 @@ describe("functions/processQueueItem.handler", () => {
     });
   });
 
+  it("pauses for rate limiting", async () => {
+    // Mock the hitrate table, but only the first three should matter.
+    mocks.scanItems
+      .onCall(0)
+      .returns(makePromiseFn({ Count: 3 }))
+      .onCall(1)
+      .returns(makePromiseFn({ Count: 2 }))
+      .onCall(2)
+      .returns(makePromiseFn({ Count: 1 }))
+      .onCall(3)
+      .returns(makePromiseFn({ Count: 1 }));
+
+    mocks.requestPost
+      .onCall(0)
+      .resolves(negativeMatchResponse)
+      .onCall(1)
+      .resolves({});
+
+    await expectCommonItemProcessed(false);
+
+    // Scan should be called 3 times to reflect pausing for rate limit.
```

Comment:
Don't really have an explicit way to assert that pausing happened, but the pause loop will run a table scan each time through to make a decision to rate limit.